### PR TITLE
Adds STM32U5 support

### PR DESCRIPTION
**Pull request by dgarske**
Thank you ST

### CODE REVIEW

**Comment by dgarske** [comment]
@danielinux , I need to still run tests on actual U5 hardware here before it is ready for merge.

**Comment by dgarske** [comment]
Note to self, customer is asking for this, so @dgarske escalate testing.

File: docs/Targets.md [hunk 1]
```diff
@@ -199,7 +200,114 @@ arm-none-eabi-gdb
...
 ```
+## STM32U5
 
+### Scenario 1: TrustZone Enabled
+
+#### Example Description
+
+The implementation shows how to switch from secure application to non-secure application.
+Thanks to the system isolation performed to split the internal Flash and internal SRAM memories
+into two halves:
```
Comment (anhu) [inline docs/Targets.md:211]: This sentence is awkward.  Perhaps you want to say: 
Thanks to the system isolation, you can  split the internal Flash and internal SRAM memories
into two halves:

File: docs/Targets.md [hunk 2]
```diff
@@ -199,7 +200,114 @@ arm-none-eabi-gdb
...
+
+The implementation shows how to use STM32U5xx in DUAL_BANK mode, with TrustZone disabled.
+The DUAL_BANK option is only available on this target when TrustZone is disabled (TZEN = 0).
+
+The flash memory is segmented into two different banks:
+
+  - Bank 0: (0x08000000)
+  - Bank 1: (0x08100000)
+
+Bank 0 contains the bootloader at address 0x08000000, and the application at address 0x08008000.
```
Comment (anhu) [inline docs/Targets.md:260]: Are both addresses supposed to be the same?

**Review by danielinux — APPROVED** [review]

**Comment by dgarske** [comment]
@danielinux over to you for final review/merge. I tested on my STM32U585I board successfully with TrustZone enabled.

**Review by danielinux — APPROVED** [review]

**Pull request merged**